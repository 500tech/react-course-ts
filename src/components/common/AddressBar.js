import React from 'react';
import { withRouter } from 'react-router-dom';

const BaseAdreesBar = ({ location, history }) => {
  const { pathname } = location;
  const { goBack, goForward } = history;
  return (
    <p>
      <span onClick={goBack}>{'<'}</span>
      <span>{pathname}</span>
      <span onClick={goForward}>{'>'}</span>
    </p>
  );
};

export const AddressBar = withRouter(BaseAdreesBar);
