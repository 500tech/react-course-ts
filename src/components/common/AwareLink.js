import React from 'react';
import styled from 'styled-components';
import { Link, Route } from 'react-router-dom';

const StyledLink = styled(Link)`
  color: ${props => (props.match ? 'black' : 'blue')};
  pointer-events: ${props => (props.match ? 'none' : 'all')};

  &:hover {
    color: red;
  }
`;

export const AwareLink = ({ to, exact, children }) => {
  return (
    <Route
      path={to}
      exact={exact}
      children={({ match }) => {
        return (
          <StyledLink to={to} match={match}>
            {children}
          </StyledLink>
        );
      }}
    />
  );
};
