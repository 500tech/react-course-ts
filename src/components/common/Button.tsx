import * as React from 'react';

interface ComponentProps {
  children: any,
  onClick: () => void
}

export const Button = (props: ComponentProps) => (
  <div className="button" onClick={props.onClick}>{props.children}</div>
);

export default Button;
