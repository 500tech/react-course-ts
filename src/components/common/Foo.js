import React, { memo } from 'react';

export const Foo = memo(() => console.log('render') || <p>Foo</p>);
