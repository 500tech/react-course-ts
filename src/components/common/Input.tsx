import * as React from 'react';

const Input = (props) => (
  <input
    ref={props.onRef}
    type="text"
    placeholder="something"
  />
);

export default Input;
