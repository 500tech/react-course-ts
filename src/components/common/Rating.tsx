import * as cx from 'classnames';
import * as React from 'react';

interface ComponentProps {
  amount: number,
  updateRating: (val) => void
}

const Rating = (props: ComponentProps) => (
  <div className="rating">
    {
      [1, 2, 3, 4, 5].map(val => {
        const classes = cx('rating-circle', {
          filled: val <= props.amount
        });

        return (
          <div
            key={`rating-${val}`}
            className={classes}
            onClick={() => props.updateRating(val)} />
        );
      })
    }
  </div>
);

export default Rating;
