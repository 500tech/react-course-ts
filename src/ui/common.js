import styled from 'styled-components';

export const RudeButton = styled.button`
  :disabled {
    opacity: 0;
  }
`;

export const Page = styled.div`
  width: 100%;
  height: 100%;
  background-color: ${props => props.theme.colors.bgColor};
`;

export const Title = styled.h1`
  background-color: yellow;
  margin: 0;
  text-decoration: underline;
  color: ${props => props.theme.colors.primary};
`;
