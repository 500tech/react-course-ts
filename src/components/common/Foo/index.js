import { connect } from 'react-redux';
import { Foo as BaseFoo } from './Foo';

function mapStateToProps(state) {
  return {
    count: state.count,
  };
}

const withFooConnector = connect(mapStateToProps);

export const Foo = withFooConnector(BaseFoo);
