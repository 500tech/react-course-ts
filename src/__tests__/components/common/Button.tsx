import * as React from 'react';
import { shallow } from 'enzyme';

import { Button } from '../../../components/common/Button';
import { noop } from 'lodash';


describe('<Button />', () => {
  test('should render hello world', () => {
    const component = shallow(
        <Button onClick={ noop }>Hello World</Button>
    );

    expect(component).toMatchSnapshot();
  });

  test('should fire on click when clicking on the button', () => {
    const spy = jest.fn();
    const component = shallow(
        <Button onClick={ spy }>Hello World</Button>
    );

    expect(spy).not.toHaveBeenCalled();

    component.simulate('click');

    expect(spy).toHaveBeenCalled();

  });
});


