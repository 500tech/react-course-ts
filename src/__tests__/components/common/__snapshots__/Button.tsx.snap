// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render hello world 1`] = `
<div
  className="button"
  onClick={[Function]}
>
  Hello World
</div>
`;
